src/System.Management.Automation/engine/regex.cs

```diff
@@ -171,8 +164,47 @@ private void Init()
                         : StringComparison.CurrentCulture;
                 }
 
-                _isMatch = str => string.Equals(str, Pattern, stringComparison);
-                return;
+                return stringComparison;
+            }
+
+            if (Pattern.Length == 1)
+            {
+                switch (Pattern[0])
+                {
+                    case '*':
+                        _isMatch = s_matchAll;
+                        return;
+                    case '?':
+                    case '[':
+                    case ']':
+                    case '`':
+                        break;
+                    default:
+                        // No special characters present in the pattern, so we can just do a string comparison.
+                        _isMatch = str => string.Equals(str, Pattern, GetStringComparison());
+                        return;
+                }
+            }
+
+            if (Pattern.Length > 1 && Pattern.AsSpan().Slice(0, Pattern.Length - 1).IndexOfAny(s_specialChars) == -1)
```

Comment:
Why not leverage the return value of `string.indexofany(char[])`?
> The zero-based index position of the first occurrence in this instance where any character in anyOf was found; -1 if no character in anyOf was found.

So we can do this:
```
if (Pattern.Length == 1 && Pattern[Pattern.Length - 1] == '*')
{
    ...
    return;
}

int index = Pattern.IndexOfAny(s_specialChars);
if (index == -1)
{
    ...
    return;
}
else if (index == Pattern.Length - 1 && Pattern[index] == '*')
{
    ...
    return;
}

var matcher = new WildcardPatternMatcher(this);
_isMatch = matcher.IsMatch;
```

Reply:
Fixed.